tensorflow_quantum/core/ops/math_ops/tfq_inner_product_grad.cc

```diff
@@ -62,8 +62,8 @@ class TfqInnerProductGradOp : public tensorflow::OpKernel {
 
     // Create the output Tensor.
     const int output_dim_batch_size = context->input(0).dim_size(0);
-    const int output_dim_internal_size = context->input(3).dim_size(1);
-    const int output_dim_symbol_size = context->input(1).dim_size(0);
+    size_t output_dim_internal_size = context->input(3).dim_size(1);
+    size_t output_dim_symbol_size = context->input(1).dim_size(0);
```

Comment:
Why remove the const ?

Reply:
In most of these cases, I looked to gcc's warning message and kept the changes limited to that. In these cases, the warnings were:

```
tensorflow_quantum/core/ops/math_ops/tfq_inner_product_grad.cc:184:53:
warning: comparison of integer expressions of different signedness:
'std::vector<float>::size_type' {aka 'long unsigned int'} 
and 'const int' [-Wsign-compare]
```

For whatever reason, it didn't mention `const` in relation to the size_type, so I didn't include it in the changes (and gcc doesn't warn about a missing const after the changes).

But now that you mention it, I went back and tried adding `const`, and it works, so ¯\_(ツ)_/¯. I'll update this PR to add const, and also check other cases where this might have happened.